crates/cyclers/code_generation/accessor.rs

```diff
@@ -0,0 +1,387 @@
+use convert_case::{Case, Casing};
+use proc_macro2::{Delimiter, Group, Punct, Spacing, TokenStream, TokenTree};
+use quote::{format_ident, TokenStreamExt};
+use source_analyzer::PathSegment;
+
+use super::reference_type::ReferenceType;
+
+pub fn path_to_accessor_token_stream(
+    prefix_token_stream: TokenStream,
+    path: &[PathSegment],
+    reference_type: ReferenceType,
+    instance: TokenStream,
+    cycler_instance_prefix: TokenStream,
+    cycler_instances: &[String],
+) -> TokenStream {
+    fn path_to_accessor_token_stream_with_cycler_instance(
+        prefix_token_stream: TokenStream,
+        path: &[PathSegment],
+        reference_type: ReferenceType,
+        cycler_instance: Option<&str>,
+    ) -> TokenStream {
+        let mut token_stream = TokenStream::default();
+        let mut token_stream_within_method = None;
+
+        let path_contains_optional = path.iter().any(|segment| segment.is_optional);
+        if !path_contains_optional {
+            token_stream.append(TokenTree::Punct(Punct::new('&', Spacing::Alone)));
+            if let ReferenceType::Mutable = reference_type {
+                token_stream.append(TokenTree::Ident(format_ident!("mut")));
+            }
+        }
+
+        token_stream.extend(prefix_token_stream);
+
+        for (index, segment) in path.iter().enumerate() {
```

Comment:
This whole loop with the Option `token_stream_within_method` looks like it could be solved cleaner with a recursive function, since you essentially want to decide where to place new tokens depending on a previous iteration.

Reply:
I tried this once but haven't found a clean way to do it while achieving the same feature set. Maybe keep it for now and add a point to Ideas for Next Iteration?

Reply:
Added to ideas for next iteration.